### PR TITLE
Mouse mri form

### PR DESCRIPTION
**Pull request by david-roper**
version 1.0 of the mouse mri form
closes issue #5 

form can be seen [here ](https://playground.opendatacapture.org/?files=NobwRAxg9gdgLgU3mAXGA9AKgAQIM4A2AlvALQAmReAhgEYELYAOCATgGYIRxGxVzo8UVnFJRaAKy5w82TOgA6MJUQC2TYXGwhs5BOxIIAkjDxxWAV1VItAX2ztWUVdgDk6S-DUJ0ANwCM6AACUCww5NRw1BDUTHAWrD6ePNak0ImuKuqa2tgAXtj2js5uHhZe1n6BeVDkQQDMAHQATE0AHpnKXdCmWngxMABy1NYA8nG8ptgAvNpK2NgKYAAyUDHEeWxLKIsra9QbW2AANPO7ACr%2BAPoAYssAggDKABJX9QAiV-4ADN9UUF9vnoAObbJaXW4PF5vT4-P5CQEgpanGALJZ3J6vD5XRLkCwQBDkMFgDHQ7G4-GE5FnJb3d4AKSuACFvgBZe4ABWJdMZLPZXJONLAiV8gyg51YRGB93IEg5AFFRux2PKOUZiSKxRKpTK5Yrlar1YLUbsWdhWbFiWaLUxqSalkz-KpYlcAEr3V3yq4ANiBCGBVwALHgrU6Xe7PT6-QHg3a0WBHc6mG6PV7-M1o0GQ6gHWHkxG0xmQVm47tWQBxL2s85K9jEitVmvK0tLBtXaujZQ5sBtjtdlHx3tN9hXc4Ad2u9Wj9cr7eHo4nb2nxvj52aZmorFuRk%2BhquNDgVwiMCuvjYV2EUpIB2Ja43W5uO6ue4PR%2BoJ7PW8vwOvBBbYAQNpfCIXwoHed9SWeYlAOA0DwJgSCliUJRbGQrpAI0ERdH0agLAILQ9AMGBjF6SxrHgAAKEAzgAaxIcgdlcG5RldVlXAHbACHfYELGoYEEEYpB2LOKJgTwHZgFcVkoAsPAEHYtxWVdIwFNcR5zHxeJWAOVT2CUlSAF0OJIRBWBgA4dmok0FkJIgeFgHZ-A4hZzOsRjWVGABVR4q2U24WLYs5bA4npEHgSyzgWVRJXGNhImEYY3LmayFlSujwh2JYzElGBQRXVKCt8TciHfOBMrAEgmAsOBSwKhYuNoBACHKxLGCgdhzWU7BQjiuBhCQlLgsi7BoCIAhzgATxYCKUoK9KGN2bKSDy5y6qKyVSvK7TKCgWq6oaprysmlhuo60a-3yuruomWBxOSq6rqWABhVgJrWKAxuJF63vOvaHt2VpVBcUCCCsBBzCICBtgB%2BogewEGwYhqGwGGgrUMG1amE3ahgLwSYADU2Dxhz7oe%2BbyqW3K-tS9aSvC3Ztt4an6roQ7dg5bHccmeGicmZnrvs0wZv%2Bgqlg5fHvWJcXJcukWxfxgBWaH5aVlHZtS9G0ZRYb%2BnfV0uGEcgTHYKBhbq8mGYN1hyFITdtIm6mDuassut11FEnSIlZYWAwmvIOSytJq7oqIR4Blas3-otrKIap721uKzbFqa6R%2BZZxrnaWMP32wVyEDTgXJjut3WvGQWQ3VhYhsr3A2nxkCoGzmBI7J%2BjytoKAoAYd8C9ppOlkZ3b44Kp3yoAdWoWQ3d0WA2tRQDSFgqBsDwCxJGkAB%2BAb-ur-7APr0CXvfXh1MiWSW6ui3XHICbXMh4Sa4WRJwjYCiIiiABKIORaIdg38iagjR94Nybh-KyIs6qJC0qicBEDzZt0WrHFaqM4F93pgPagO0C5XVHhcKabUOoLyXivAY2C6qhHLhfOBoswAmDSNpGATNux0IgAwphq1qE0PlG0ehx8h7lW4bwxhQ8UEQM1pw8RYjRGQPBgkVEMA8J-kfoUaRu8Hp6DaJ5OSC1YFRwQUsDuXcEA92HjTRO6DhSYPYdI9ObMlgT1kO8QC1g9B9VUEQSgxFsCyUJFvNWO9tY1w0Y8Lu1VJjgUQFQua%2BiwA3zvsjDhMiX6sH-p-b%2B-1f6pMARorRhIwE2JkdA9JcDo6xMiPnUx-1cFLCcW0Fx4NnAeMMNgEJoNBa6HKWQquBTunKKfrIsyudFEFMkXVau0iNFMkiBAAAFoMKwjVWA7GKdEjKuw4kjEhgXZ%2BegUnv2oF-XRP8-77MaDk7R%2BS%2BmpSgXIlZItSmU2QVcwq5jA5LEqtVLpqVqlgFqfUtxTSvG0GmTMoZqhFlfNGTvHp2AbmDIUQQJRUia5qP%2Big-47BQbCHfAgXJOjpGlMMd3fs0i0FaC2lYoeiTbGZ1oUITFFhsVeJ8eQPxKD7DUsWOrDFWKGEICmXAWZ8zwVsCidgUpGz3EJOkTs1%2B%2BzLkQMyacnljK%2BV4oVXAuFMCYVpQQa4R5D9OHw1eeVD5NVKn7VZrSow9LeU4uwMCwVoKFEitYNvOBUKxkFK1UMxFqj2WJJQei21qqcW6lkogfFj8JW302dKmusq9kAI1Rkk5ADGgqqZbii5Rz-o%2BtzSUvVRLjEwENUa41G0LGDy%2BTS8qNr2p2q8ZgiQEbCQOCcC4R4owORsquZ6tG3qBnyOGSigNaErqZr5eGswhIORsAJPAPiAk7kxvifzRNWTDkFKVemydYbZStvICmkW%2BadXipiY8r5ZLTUwCquazlOCrV1pDVm5g86bBLrIf2jWg6ikIqRQ9T14z1aqBknJeuUQCB4AlNEGihIxWEs7sS-m16GaUv5qPexbBGBwBmYwY%2BzoCD6vhnZA4shzCwbbXiHKwISE5woggRowJGgtI5KMZoxxsDsZGrABdFHBacdoIkSIMzlrHHBhARoH9e1XWCo9dWQnjG4eWqfegy6C0LCvpK%2B%2BnKN3yruQsHdURGigZ8RBsjMGIBwaPRpwptzbOXxicWkxD6E6VreZYrBFqR5Pt2A4h1wnlO5RXlENTMmPUjN-bc-9-r1bAaulANoE1%2BIwCerxmw2lBYIb1dpiAZarp6eTQZ7ARnAGmfA6RqDlnrPHrzUO4rqyFpLBdRC7zZj3M3rvV8n5owkspewGlmAfHMvcwot8UgcIACkH93XIrmye%2BrMXR1xcSYl5LSBHhnxGyTWzWnY1Svy0k3Zm7iulZM2BhA5mqvaSs3khz1z6v3ca%2BVFrRxXMvI67sM13XfNZzY80Y1oNGBjYm78abs2Rbft6RAn1S3oUrZQYkPATAiDbZgK6cp2W1nX32zpmVSBjv6fu2d8rl3KvQZuzVp7sLHtns0zE17bq2sfbph577zPvm-bAPrZHqPIjc0y8DxTIm8DoHcTAGb-iIvKKhzTv9I74eycCVdE2rBVBKlZMY1eiRyKBxXTl3HeXdME7lUV4nabjOk6uxTyjNmYWnuefAtZMcaM1va6zimKduBu9rbsZ4UAxzYDHNhrjfWkA8aGxl-nsBOOPH%2B5x985Aae87R7C8pNPPa%2BJ94XW6YrqH2JxvoYQqhlZgAnmeVXJeOePTAMKtgkMbyoF2HXyU6xs%2BDnfLxC6KBWyd8b3Tn9fbIvKNhwrwDY71YoPYCHetDKs3Vfg-r53YBnMkprqhjBXnOU-On5KErL6%2BWcQQGeKD2AKO3dZRDqugb1a76ILPxt6mCUxNy%2Buk3SbP7m6yY0O-D-Q3EQXztyuQdyNVKUZyvRNUWmIF2R%2BwznKj0i6j-1fRYFYD4yXU4x6Dxl2TP1E1kEnhwMYCKiBwdSL0SBKy0CoF0BAioDbVoAmmwB%2BE4yICARY38AVhmAYMaAVkmy%2BWdDaEcgVne1SnFx2G%2BBhVl1l1Hz9WWyVynxnwPxxTSwIBkiWSXya1iUNzf2SU3S-1OV-wUK8WmCMOwGaFqwehAM4QeSQQgM%2B032sUdx%2BSQMP2gGUISB4z0C%2BQoSLjzweiWAmiamULHFLwAE0AiA83clhgREgkBS9yxoiuwz1ZdodIcot4Ux9ZNhpUVUooBcM2A0sgYbA7pdsL1rDTEN8wBEA2gohhMMMucORu45Ig9JREBsB3x6DMFKA%2BpzICAeMCj4BRcYAcj8BsA8NEgr8VE4tMiOI1FXFqAxoijho9B%2BhJQbpm4Lhl4PZDYSsYBK9o9UQ6AZItBqBsBSdiN9I6NUQ5I8BiYYAWMABpBABAJgcjSnU6E4vCHgJgBgC42QMcOyUTfYleZab4q4m4ljJ6HORqBwMaBgJPEgB1eghAf4tgTqIwa6XqYQbqVgC4mnAARwsHwFMkaD2kJLUHKXIHeASD2MclWhIGyk0m8OwEklyRwMoMrwcExJznomoLxAOBOIu1RJXnwBuLwE4zHFE1mRGhzlCjmNRFUA%2BKIC%2BMYDdlkHaioOVGw3gHPXCDwBYyMAoNkAwmkDbVw0iAILP3wTeOnlNKOLIPIFnm6lRFyP5J8VaN%2BICLdJGjYCiHhJIF2PaVXklPwOF1wzT0QAT3CG6jD3MnaQYBP3bRKGdPONA0YT6mxNVy9JEFlJ%2BJYzHjw3ditk8Vo2dJVOYAaMYDvQIJcHhOdMAmiC0ENhRNyPoODztIdL%2BKC3NNBMmEaFcCMmGmIAXTkkYk8kGGWCMCenlEGB8neEOx4DgAYHcgFPOJuGL06GvzOGsEngSHwBbhDlikywShGCf3VlKUwPvRQR%2BXOJ6kPKZ0SUSHYHKn3JQPilYFaghyyPOmOhPMc2x3PMO191cDSzGgtJYAApp0fLcC-PwXXI1kxk5ioAJl5h2xQSvn-MSVwVcA5m0i5lgGwEJlYBuPAofMYixhwsQtgAIqIqmJ1gGH1k9mNlNmKTPOcDvSpHjh%2BSbgzy2L9KpRQSIIJJ2BO2mAAD47lzzAcODTk3Z6LDZGKN4TNYgKIKI2gv5RLsBVLpEGAtAtztcEBRhqpKzZhXBYKJ0-4iCzCCoMyKIJLeK3iLKGsTitcdyDK4BKyABqWYXikzSUJuVqbAdytwIKwK7ACiby4BQ%2BNhRuKIeIWQDeDeIK1wL%2BQK1wYK0K7y4JUJQWCJRgBK6%2BQCFeLK7md%2BZdVKwKjKwCVpMJWAHKlAEyr%2BAKxKxqsKnYqAM5QCAVIVBZFEvKjRB1EFMFRZRiRqiqtoTquZbqpZeqgKmFVKsq9K1qjNAw-lEFYVRZbAPKvdIFAaxnYa8qxaralap1Na0VaalKtKlqk2Jahtf-BAadSNOdVAj9fiDatwJw%2B1ZtQ9N9J6xdfiParlahbyw6%2B62dd9X6xgFKybVwOqpKxqua5q7ykMgE4EVTb4vKpG5aELOgRctKxGwLZG1G0q2G86sqmFS6tqtbFLQbYbPY161wbjHoGmrLXGxaympAamqPdpSG1KxiYmpqwK8mxoNmmATbLSWmvK7jA8Kk5m%2Ba7y4W0W6W7mbm7AXm5K2ai67ypHFHNHDHFovKrWvndpQXf6zW-AbWvY3WiGtwJGsXEgHmtwVK5K-mhaq6yvDXZynXGwOm9ktU3Snc3XE2xat2vSD2hAAOh2tWq5eGgW7y-Qm619PKhAtEw676tAv6jWoO%2BQ%2BOw-dABg74J21wKG1WuGjO12rOufPlJQlQ7284lOlwlQwOsuyUd64iKuhIYuwK9W%2BawWnI0Y-I3XOKqC5wAexuim3I1gfuwojutwBQJQVwAAbmH0V3MPqz9sSFcq60mMGiCg4iII8T2LDjw2dB2DyCFvXm4ComGmfIxLfOPJPsaEeQog-lWmgumnyAfqQSfvgvIpuKosmHvsfuftor1kLMYvvrtmoAmgolPvECkAvoLRDj8rvvfsAcSQisbgGHvtXy-pQXQaPmEVPlioAc-o-iFtWIOBwfVnOUX1PuwaAaocqqKpqsx3fpKqfrIcFgofoaukmVWsmuIZo3Ya8NgC4cSUOrxSwaQxLUoYnWWvGpOtUNPsAY4cmFEeDWzv3RbRnQWloakffCEfIYIBkbqmBoPW0cerTuXSUZIZUZEaMe4bqit3J0AMkaMX0YcYKgxtykJtceJQMc4fsdW2jI5vgDR3vsZ3YfFzG1Ib4IojhFIeEe6OMYKnlq22pPfoiZiZIGicUraDid%2BAScMeSYexTwtpYdPsyZM2yfztyYonqAVhqcSbUdv2L3du3M9vpmscEcKYCeKZ9nLsfxcffrocSTjorpxXCcmv8dUcCbkObuWrbsUY-u6dsaSY8eyPHsnv6IEeWmmbsafvHQ1g-gcdsBmxgFOfnpODADzlQAqhfjaEaBkDAFsAMiAA&label=LYewrgzgpg%2BsBOBLGAzE9hA)

### CODE REVIEW

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        kind: "string",
+        variant: 'select',
+        label:"Type of MRI done",
+        options: {
+            "Ex-vivo structural":"Ex-vivo structural",
+            "In-vivo structural": "In-vivo structural",
+            "Structural and FMRI": "Structural and FMRI",
+            "Quantitative":"Quantitative"
+        }
+    },
```
[the remark this answers is not in the document]
Comment (gdevenyi) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts]: Yes

[the remark this answers is not in the document]
Comment (gdevenyi) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts]: Well, yes and no, I want the record array, but I want the sequence choices to be far more specific, and obviously each should be seperate.

Also add to description to put them in the order they were done

[the remark this answers is not in the document]
Comment (gdevenyi) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts]: Please talk to dan about how to they document/specify sequences.

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+      label: "Respiration rate (breaths/min)"
+    },
+    (type) => type === "Structural and FMRI" || type === "Quantitative"),
+    formOfMeasurement: createMRIDependentField({
+      kind: "string",
+      variant: "select",
+      label: "How were Oxygen concentration, SPO2, and respiration rate recorded?",
+      options: {
+        "Waveform": "Waveform",
+        "Numerical": "Numberical",
```
[the remark this answers is not in the document]
Comment (gdevenyi) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts]: > Numberical

I don't think this is a word.

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  tags: ['Mouse', 'MRI', 'Structural', 'fMRI'],
+  internal: {
+    edition: 1,
+    name: 'MOUSE_MRI_FORM'
+  },
+  content: {
+    handlerName: {
+        kind: "string",
+        variant: "input",
+        label: "Name of handler"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:32]: isn't this the person filling out the form? If so, this isn't needed.

**Review by gdevenyi — CHANGES_REQUESTED** [review]

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (david-roper) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts]: oops

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        variant: "input",
+        label: "Name of MRI operator"
+    },
+    coilType: {
+        kind: "string",
+        variant: "radio",
+        label: "Type of coil",
+        options: {
+            "Cryocoil":"Cryocoil",
+            "23m volumetric": "23m volumetric"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:40]: pretty sure the coil isn't 23 meters in diameter.

**Review by gdevenyi — CHANGES_REQUESTED** [review]
For the MRI records, I want
in-vivo/ex-vivo
type: structural, functional, quantitative, other
sequence: <contents dependent on type, contains a list of all sequences we use>
comments:

the contents of sequence needs to be coordinated with Dan

**Comment by david-roper** [comment]
Notes from meeting with Dan:

Type of mri to be renamed to MRI protocol
add more description to addition comments section
- note if scan had to be redone, etc.
Add a new question for scan type dependent on protocol
- ex: neonates or workhorse for structural protocol
Add a new question contain possible scan/sequence number (Dan will provide a list for these)

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+            }
+          }
+          return null
+        }
+      },
+      
+      fmriIsofluoraneTracked: {
+        kind: "boolean",
+        variant: "radio",
+        label: "fmri isofluorane levels tracked?"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:253]: fMRI

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+          kind: "string",
+          variant: "select",
+          label: "Scan name",
+          options: scanNameOptions
+        },
+        exVivoScan: {
+          kind: "boolean",
+          variant: "radio",
+          label: "Was scan done on ex-vivo subject?"
+        },
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:72]: this is a subject-level, not a scan-level question

in/ex vivo and in/ex craino only need to be asked once at the top

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+            }
+          }
+          return null
+        }
+      },
+
+      mouseVitalsTracked: {
+        kind: "boolean",
+        variant: "radio",
+        label:"Were the animal's vitals tracked during scan (e.g. SPO2, O2 concentration, breathing,etc.)?"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:175]: subscript or at least underscore for the formatting of SPO_2 and O_2

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+        variant: "input",
+        label: "Name of MRI operator"
+    },
+    coilType: {
+        kind: "string",
+        variant: "radio",
+        label: "Type of coil",
+        options: {
+            "Cryocoil":"Cryocoil",
+            "23mm volumetric": "23mm volumetric"
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:45]: this is not all coil types, check with Dan.

"23 mm", ISO formatting for units says to have a space.

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts [hunk 9]
```diff
@@ -0,0 +1,368 @@
+/* eslint-disable perfectionist/sort-objects */
+
+import { defineInstrument } from '/runtime/v1/@opendatacapture/runtime-core'
+import { z } from '/runtime/v1/zod@3.23.x'
+
+
+const scanNameOptions = {
```
Comment (gdevenyi) [inline lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:7]: you ask PV5/PV6, do you have all the old PV5 sequences that might be used?

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts (hunk 6 above)
Comment (gdevenyi) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:72]: All the physiology monitoring and anaesthetic can be suppressed if this is ex-vivo scanning.

**Review by gdevenyi — CHANGES_REQUESTED** [review]

File: lib/forms/CoBrALab-Mouse-MRI-Form/index.ts (hunk 9 above)
Comment (david-roper) [reply lib/forms/CoBrALab-Mouse-MRI-Form/index.ts:7]: from the list dan sent the only pv5 protocol still used is the Ex-vivo protocol, which includes scan names 01-Tripilot-multi and exvivoDanFLASH

**Review by gdevenyi — APPROVED** [review]

**Pull request merged**